src/sentry/api/serializers/rest_framework/sentry_app_installation.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import absolute_import
+
+
+from rest_framework import serializers
+from rest_framework.serializers import Serializer, ValidationError
+from sentry.constants import SentryAppInstallationStatus
+
+
+class SentryAppInstallationSerializer(Serializer):
+    status = serializers.CharField()
+
+    def validate_status(self, new_status):
+        # make sure the status in in our defined list
+        try:
+            SentryAppInstallationStatus.STATUS_MAP[new_status]
```

Comment:
so I don't think we necessarily need the `STATUS_MAP` - if we only allow for updating to be installed we can probs just do that explicitly

```python
if new_status != SentryAppInstallationStatus.INSTALLED:
    raise ValidationError(u"Invalid value '{}' for status".format(new_status))
```

That way do have to worry about the check below since we don't let anyone explicitly set an installation's status to `pending`

Reply:
@MeredithAnya I talked to @mnoble about this and he said we should let our APIs be idempotent (see https://restfulapi.net/idempotent-rest-apis/). In other words, if the installation is `pending` we shouldn't throw an error if the request sets it to `pending`.

Reply:
> They can write client code in such a way that there can be duplicate requests as well.

To me that's not what we are saying here imo. It seems like pending is a state that is never set directly by the developer. Like they should never be trying to set the status to pending.

Reply:
@MeredithAnya I am OK with either approach but I'd like to know what @mnoble thinks before making the cahnge